### ISSUE TITLE
Git isn't working as expected with vscode remote ssh

### ISSUE THREAD

**Issue by rjmccluskey** [microsoft/vscode-remote-release#1094]
I recently tried out VSCode's remote SSH feature. It was super easy to setup and connect to a remote server. 

I opened the editor to a directory that is a git repo and expected to have the default git features working like normal. But if I edit a file, VSCode isn't showing the visual indication that there is a modification in the file. And the source control tab doesn't show the change. I can open the terminal in VSCode and run `git status` and I do in fact see the change so I know that's actually editing the file.

If I ssh into the server and run `git status` I do notice that VSCode updates and now shows the change in the source control tab. But there's still no visual indication next to the line number in the actual file that is changed.

**Comment by joaomoreno** [comment]
> But if I edit a file, VSCode isn't showing the visual indication that there is a modification in the file.

In the SCM viewlet, there is a Refresh button at the top. If you click it, do the changes show?

**Comment by rjmccluskey** [comment]
@joaomoreno If I click the refresh button then the changes will appear in the "Changes" list, but I never see an indication in the actual file that I edited. Normally, when using vscode locally, you see a visual indication next to the line number like this:

![Screen Shot 2019-08-07 at 1 51 23 PM](https://user-images.githubusercontent.com/8008311/62657073-771a2080-b91a-11e9-8f9c-402ccb3caf90.png)

The reason I'm asking about all this is because I'm trying to evaluate if I want to use remote ssh for my work and have other devs potentially use it. Without getting into a bunch of detail, we can't run our code locally but instead edit files on a remote server which has our development environment setup. Right now I do this by using rsync. I was hoping that using remote ssh would be a lot simpler but without the automatic git features of vscode I'm not sure if it's what we really want.

If that's not on the roadmap for this feature or not the intended use case then no big deal, I was just curious if I was doing something wrong.

**Comment by joaomoreno** [comment]
This should work. In fact, it works for me.

Can you show me what appears in `Git: Show Git Output`?

**Comment by rjmccluskey** [comment]
Here's my output:

```
Looking for git in: /usr/bin/git
Using git 2.17.2 from /usr/bin/git
> git rev-parse --show-toplevel
> git rev-parse --git-dir
> git config --get commit.template
Open repository: **[path redacted]**
> git rev-parse --show-toplevel
> git status -z -u
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git symbolic-ref --short HEAD
> git rev-parse **[branch name redacted]**
> git rev-parse --symbolic-full-name **[branch name redacted]**@{u}
> git rev-list --left-right **[branch name redacted]**...refs/remotes/origin/**[branch name redacted]**
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> git remote --verbose
> git check-ignore -v -z --stdin
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
```

FYI, I removed some sensitive info, paths and branch names.

**Comment by rjmccluskey** [comment]
And when I click refresh, here's the output:

```
> git status -z -u
> git symbolic-ref --short HEAD
> git rev-parse [branch name redacted]
> git rev-parse --symbolic-full-name [branch name redacted]@{u}
> git rev-list --left-right [branch name redacted]...refs/remotes/origin/[branch name redacted]
> git for-each-ref --format %(refname) %(objectname) --sort -committerdate
> git remote --verbose
> git check-ignore -v -z --stdin
```

But as I am actively editing and saving files there is no output. If I switch to a different file then I get...

```
> git rev-parse --show-toplevel
```

But it doesn't result in showing any changes in the source control tab.

As mentioned before I still never see visual indication of changes in the side of the file next to the line numbers.

**Comment by joaomoreno** [comment]
Alright can you run `Developer: Set Log Level...` and choose `Debug` and then capture the git output?

**Comment by rjmccluskey** [comment]
I can't unredact the path, but I took another look at it since you mentioned that. Turns out it was a different path then the directory I selected when vscode prompted me to pick a directory to open.

When I SSH into the server I end up in somewhere like `/home/username`. So vscode prompted me to pick a directory and I picked `/home/username/repository`. These paths aren't the exact naming but you get the idea.

That seemed to work just fine, but the git logs showed that it was using a repository path of `/mnt/ebs/home/username/repository`. I think this is where things got messed up.

If instead I tell vscode to open to `/mnt/ebs/home/username/repository` then everything works as expected! 🎉

I wonder if this is a quirk of using EBS as the remote server? I'm still getting familiar with how our remote server is setup and I may need to ask people about that. I think for now this should work for me.

Thanks for taking the time to help me out with this. I'm sorry if it was frustrating.
